### ISSUE TITLE
Random seed not implemented for Orthogonal Initializer

### ISSUE THREAD

**Issue by cosminc98** [tensorflow/tfjs#7376]
**System information**
- TensorFlow.js version: 3.17 (but it isn't implemented in any of them)
- Are you willing to contribute it: Yes

**Describe the feature and the current behavior/state.**
For reference, if you try to use the Orthogonal initializer with a seed you would get the following error:
`Random seed is not implemented for Orthogonal Initializer yet`

Current implementations:
- [Python](https://github.com/keras-team/keras/blob/v2.11.0/keras/initializers/initializers_v2.py#L752-L770)
- [JavaScript](https://github.com/tensorflow/tfjs/blob/tfjs-v4.2.0/tfjs-layers/src/initializers.ts#L546-L567)

The current Orthogonal initializer in TFJS does not use a seed (which is needed to reproduce results across multiple training sessions) to initialize the RNG. Also, the TFJS version uses [tf.linalg.gramSchmidt](https://js.tensorflow.org/api/latest/#linalg.gramSchmidt) instead of [tf.linalg.qr](https://js.tensorflow.org/api/latest/#linalg.qr) like the Python version (this is mostly a matter of consistency). There are two main modifications to be made:
1. Add seeded RNG (and tests for it)
2. Replicate the Python implementation for consistency sake.

**Will this change the current api? How?**
The current API does not change. The signature of the [apply](https://github.com/tensorflow/tfjs/blob/tfjs-v4.2.0/tfjs-layers/src/initializers.ts#L546) method stays the same. Only the implementation itself and some tests need to change.

**Who will benefit with this feature?**
People training models with weights initialized by the Orthogonal initializer as the method of generating the values matches the one from python (it should thus be easier to replicate results from python) and is reproducible given a fixed seed (should allow for less variability between training sessions by starting with the same weights which is useful for research / debugging).

**Comment by cosminc98** [comment]
A similar issue I have just discovered: https://github.com/tensorflow/tfjs/issues/5955

**Comment by mattsoulanille** [comment]
Fixed by #7377. Thanks for the PR!
